### PR TITLE
probe: move hardware related root node data to a probe

### PR DESCRIPTION
**Pull request by safchain**

### CODE REVIEW

**Comment by safchain** [comment]
run skydive-functional-tests-backend-orientdb 
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-orientdb
run skydive-functional-tests-backend-elasticsearch

File: topology/probes/hardware/hardware.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+// Start the probe
+func (p *Probe) Start() error {
+	p.Ctx.Graph.Lock()
+	defer p.Ctx.Graph.Unlock()
+
+	tr := p.Ctx.Graph.StartMetadataTransaction(p.Ctx.RootNode)
+	defer tr.Commit()
+
+	if instanceID, err := getInstanceID(); err != nil {
+		p.Ctx.Logger.Error(err)
```
Comment (lebauce) [inline topology/probes/hardware/hardware.go]: I'm not sure we should log this as an error as the presence of the /var/lib/cloud/data/instance-id file is not the most common one. So you should probably use `Info` or even nothing if the file is missing

Comment (safchain) [reply topology/probes/hardware/hardware.go]: I will use debug, so that we can still have the information if needed

Comment (safchain) [reply topology/probes/hardware/hardware.go]: I finally added something to ignore "no such file" error

**Pull request merged**